samples/web/content/apprtc/index.html

```diff
@@ -28,15 +29,36 @@
     <video id="remote-video" autoplay></video>
     <video id="local-video" autoplay muted></video>
   </div>
+  
+  <div id="landing">
+    <h1>AppRTC</h1>
+    <p id="instructions">Please enter a room name. Room names are shared with everyone, so think of something unique and send it to a friend.</p>
+    <div>
+      <input type="text" id="roomId"/>
+      <button id="joinButton">Join</button>
+      <button id="randomButton">Random</button>
+    </div>
+    <div id="recentRooms">
+      <p>Recently used rooms:</p>
+      <ul id="recentRoomsList"></ul>
+    </div>
+  </div>
 
-  <footer>
+  <footer id="footer">
     <div id="sharing">
-      <div id="room-link">Waiting for someone to join this room: <a href="{{ room_link }}" target="_blank">{{ room_link }}</a></div>
+      <div id="room-link">Waiting for someone to join this room: <a id="room-link-a" href="{{ room_link }}" target="_blank">{{ room_link }}</a></div>
     </div>
     <div id="info"></div>
     <div id="status"></div>
   </footer>
 
+  <script type="text/javascript">
+  var initialParams;
+{% if not landing_page %}
```

Comment:
When you access /r/<roomid> directly, you get the same page as before with params already populated, and initialize() is called immediately.